sql/api/src/main/scala/org/apache/spark/sql/catalyst/parser/DataTypeAstBuilder.scala

```diff
@@ -126,7 +133,14 @@ class DataTypeAstBuilder extends SqlBaseParserBaseVisitor[AnyRef] {
       val endStr = ctx.to.getText.toLowerCase(Locale.ROOT)
       val end = DayTimeIntervalType.stringToField(endStr)
       if (end <= start) {
-        throw QueryParsingErrors.fromToIntervalUnsupportedError(startStr, endStr, ctx)
+        val intervalInput = ctx.getText()
+        val pattern = "'([^']*)'".r
+
+        val input = pattern.findFirstMatchIn(intervalInput) match {
+          case Some(m) => m.group(1)
+          case None => ""
+        }
```

Comment:
could you put the parsing error code into `fromToIntervalUnsupportedError`.

Reply:
Updated. Thanks!